db/db_impl.cc

```diff
@@ -6539,9 +6539,35 @@ Status DBImpl::IngestExternalFile(
   // Cleanup
   ingestion_job.Cleanup(status);
 
+  if (status.ok()) {
```

Comment:
Actually all files are ingested in one atomic batch (everything is copied first) and then atomically we ingest all of them at once.